### PR TITLE
Fixes regression generating func.yaml. auto detected runtimes weren't…

### PR DESCRIPTION
**Pull request by treeder**
… added to file.

Closes https://github.com/fnproject/fn/issues/608

### CODE REVIEW

File: init.go [hunk 1]
```diff
@@ -34,15 +33,15 @@ import (
 
 type initFnCmd struct {
 	force bool
-	funcfile
+	ff    *funcfile
```
Comment (treeder) [inline init.go]: Made this explicit, was hard to follow what was going on previously.

File: init.go [hunk 2]
```diff
@@ -136,29 +135,31 @@ func (a *initFnCmd) init(c *cli.Context) error {
...
+	err = a.buildFuncFile(c) // TODO: Return LangHelper here, then don't need to refind the helper in generateBoilerplate() below
 	if err != nil {
 		return err
 	}
 
-	runtimeSpecified := a.Runtime != ""
+	runtimeSpecified := a.ff.Runtime != ""
 
-	if runtimeSpecified && a.Runtime != funcfileDockerRuntime {
+	// TODO: why don't we treat "docker" runtime as just another language helper? Then can get rid of several Docker
```
Comment (denismakogon) [inline init.go]: good point, been using `runtime: docker` for binaries, so it's really a nice-to-have thing.

File: init_test.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+	if err != nil {
+		t.Fatalf("ERROR: Failed run `init` command: err: %v", err)
+	}
+
+	ffname := "func.yaml"
+	b, err := ioutil.ReadFile(ffname)
+	if err != nil {
+		t.Fatalf("could not open %s for parsing. Error: %v", ffname, err)
+	}
+	fmt.Printf("FUNC: %+v\n", string(b))
```
Comment (denismakogon) [inline init_test.go]: `t.Logf` is better

File: init_test.go [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+	"github.com/urfave/cli"
+	yaml "gopkg.in/yaml.v2"
+)
+
+func TestInit(t *testing.T) {
+
+	testname := "test-init"
+	testdir, err := ioutil.TempDir("", testname)
+	if err != nil {
+		t.Fatalf("ERROR: Failed to make tmp test directory: err: %v", err)
```
Comment (denismakogon) [inline init_test.go]: better to use `err.String()`, not sure that `%v` would give human-readable err representation.

Comment (treeder) [reply init_test.go]: It does.

**Comment by treeder** [comment]
Removed println's
